### PR TITLE
Refactoring get method, add test for raise when block no given

### PR DESCRIPTION
**Pull request by shindakioku**

### CODE REVIEW

**Comment by shindakioku** [comment]
@yegor256

**Comment by shindakioku** [comment]
About raise "A block is required"

Do you really need the custom message?

You can remove this check and my test (add test for raise when block no given) will be succesffully execute but with some edit:

> assert_equal(error.message, 'A block is required')

To

> assert_equal(error.message, 'no block given (yield)')

File: lib/zache.rb [hunk 1]
```diff
@@ -52,13 +52,12 @@ def initialize(sync: true)
...
   def get(key, lifetime: 60 * 60)
     raise 'A block is required' unless block_given?
-    if @sync
-      @mutex.synchronize do
-        calc(key, lifetime) { yield }
-      end
-    else
-      calc(key, lifetime) { yield }
-    end
+
```
Comment (yegor256) [inline lib/zache.rb]: @please, remove empty lines from method bodies, see: https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

**Comment by yegor256** [comment]
@shindakioku looks good, see my comment above

**Comment by yegor256** [comment]
@shindakioku I'm not sure it's more readable now: https://twitter.com/yegor256/status/1060771538448859136

File: lib/zache.rb [hunk 2]
```diff
@@ -76,13 +72,9 @@ def exists?(key)
...
-    if @sync
-      @mutex.synchronize do
-        @hash.delete(key) { yield if block_given? }
-      end
-    else
-      @hash.delete(key) { yield if block_given? }
-    end
+    block_lambda = -> { @hash.delete(key) { yield if block_given? } }
+    return block_lambda.call unless @sync
+    @mutex.synchronize { block_lambda.call } if @sync
```
Comment (007lva) [inline lib/zache.rb]: `if @sync` seems unneeded here due to the condition from previous line.

**Comment by aleksandr-kotlyar** [comment]
```
def get(key, lifetime: 60 * 60) 
  raise 'A block is required' unless block_given?
  calc_lambda = -> { calc(key, lifetime) { yield } } 
  if @sync 
    return @mutex.synchronize { calc_lambda.call }
  calc_lambda.call
```

**Comment by yegor256** [comment]
@shindakioku I think the right solution would be to use DI for this coe

**Pull request closed without merging**